# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/GraphDriver.h
# Commit: 0ae30eae800d
# Full Hash: 0ae30eae800dcad73d997f9f12bb56eb537ef86c
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1651745
# File Overlap Count: 1
# Description:
#   Bug 1651745 - Don't buffer more than necessary in AudioInputProcessing. r=padenot
#   
#   When opening a second input track, there will already be some data from its
#   first instantiation in the driver's scratch buffer. If we ignore this data, we
#   end up buffering too much in AudioInputProcessing::Pull and tripping an assert.
# ==============================================================================

diff -r a00ad7266f54 -r 0ae30eae800d dom/media/GraphDriver.h
--- a/dom/media/GraphDriver.h	Thu Nov 05 16:43:24 2020 +0000
+++ b/dom/media/GraphDriver.h	Thu Nov 05 16:44:00 2020 +0000
@@ -185,7 +185,8 @@
   /* Called on the graph thread when there is new input data for listeners. This
    * is the raw audio input for this MediaTrackGraph. */
   virtual void NotifyInputData(const AudioDataValue* aBuffer, size_t aFrames,
-                               TrackRate aRate, uint32_t aChannels) = 0;
+                               TrackRate aRate, uint32_t aChannels,
+                               uint32_t aAlreadyBuffered) = 0;
   /* Called every time there are changes to input/output audio devices like
    * plug/unplug etc. This can be called on any thread, and posts a message to
    * the main thread so that it can post a message to the graph thread. */